### ISSUE TITLE
.NET 5 vs .NET 6 breaking change?

### ISSUE THREAD

**Issue by vokounjan** [ikvm-revived/ikvm#44]
Hi, this is more of a question than an issue.

I succesfully built a .NET application that relied on a library that was available in Java only. I transpiled the library using IKVMC, then referenced the `net_core_compat` binaries and it worked like a charm targeting firstly .NET Core 3.1 and then .NET 5.

Upgrading the framework from 3.1 to 5 was seamless, required no action regarding IKVM and everything was working just fine so I believed that the framework upgrade from 5 to 6 would be no problem. However after upgrading my .csproj and Microsoft nugets, I'm receiving these runtime errors:

`TypeInitializationException: The type initializer for [name of a class] threw an exception.`

`NoClassDefFoundError: Could not initialize class java.nio.charset.StandardCharsets`

I have to say that I'm quite surprised by this. I've read some blog posts describing changes and optimizations between .NET 5 and .NET 6 but I still have no idea what could be the root cause of my problem and how to solve it.

Has anyone seen anything similar? Does anyone have any tip to try?

Thanks in advance. 🙂

**Comment by NightOwl888** [comment]
Out of curiosity, does the behavior change if you [revert to NLS](https://docs.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#use-nls-instead-of-icu)?

**Comment by vokounjan** [comment]
Interesting idea but unfortunately not. Changing the setting didn't make any difference but I'll probably need to read more about this topic though. Thank you for your help.

**Comment by NightOwl888** [comment]
Hi there. I noticed that each of the IKVM.OpenJDK.*.dlls weren't being generated with a `TargetFrameworkAttribute`, and I suspect that might why you are getting this error. [.NET 6 is supposed to go into compatibility mode when it loads a DLL that targets .NET Core 3.1](https://docs.microsoft.com/en-us/dotnet/standard/frameworks#supported-target-frameworks), then it wasn't because we didn't specify the framework it was targeting.

Please check .NET 6 support against #52 and let us know if this fixes the problem.

**Comment by NightOwl888** [comment]
I should ask, did you compile IKVM in to target .NET 6 (which we haven't tested at all), or did you compile IKVM using the .NET Core 3.1 target, transpile your assemblies in .NET Core 3.1 and then load them with a .NET 6 entry point?

**Comment by vokounjan** [comment]
Hi, thank you for your response. 🙂

I'm sorry that I forgot to mark this issue as resolved and post my solution - it looks like I was (after almost 2 months) able to overcome this problem (even though I'm not really sure how and why but it may very possibly be related to the link you posted). 

As I mention earlier I was using libraries from the `net_core_compat` branch (which targeted 3.1 I believe) because there was no "easy" way to build IKVM when our project started (~2 years ago). But things have changed (thanks to you and all other guys working on this project), I finally decided to give it a try and build IKVM myself and the problem was gone. 🙂

The only thing I had to do was to replace IKVM .dlls and slightly modify the IKVMC transpilation command. Previously I just called it like this:

`ikvmc -target:library -out:{target} -version:{version} *.jar`

and after upgrade I have to reference some System libraries:

`ikvmc -target:library -out:{target} -version:{version} -r:System.Runtime.Extensions.dll -r:System.Runtime.dll -r:System.Threading.dll -r:System.Threading.Thread.dll *.jar`

But that was pretty easy to fix as the compiler was quite helpful and told me what was missing.

TLDR: Using current version (built locally from master) resolved my problem. I should probably have done it earlier and before creating this issue. Thanks again to all of you supporting and maintaining this project.
